cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -734,7 +900,85 @@ impl LowerBackend for X64Backend {
                 Opcode::Jump | Opcode::Fallthrough => {
                     ctx.emit(Inst::jmp_known(BranchTarget::Label(targets[0])));
                 }
-                _ => panic!("Unknown branch type!"),
+
+                Opcode::BrTable => {
+                    let jt_size = targets.len() - 1;
+                    assert!(jt_size <= u32::max_value() as usize);
+                    let jt_size = jt_size as u32;
+
+                    let idx_size = ctx.input_ty(branches[0], 0).bits();
+
+                    // Zero-extend to 32-bits if needed.
+                    // TODO consider factoring this out?
+                    let idx = if idx_size < 32 {
```

Comment:
It seems in some places the backend measures size in bits, while in others (e.g. size field in a divide VCode instruction) it is in bytes. Might be good to standardize on bits (given that all the types, e.g. `I8`, are named as such), and use the enum suggested above here as well; it could have a `from_bits` as on `InstSize` in aarch64.

Reply:
Or have a naming scheme that makes the units explicit?  For example, `offset_szW` -- offset in words.  `offset_szB` -- offset in bytes.  `offset_sz` -- offset in bits.  Also, useful variants -- `offset_szBlg2` -- log 2 of size in bytes, etc.